app/server/apps/users/routes.js

```diff
@@ -1,28 +1,12 @@
 'use strict';
 var auth = require('../../middleware/authenticate'),
-    users = require('../../middleware/users'),
-    permissions = require('../../middleware/permissions'),
-    send = require('../../middleware/send');
+    routeGenerator = require('../../route-generator'),
+    methods = require('./methods'),
+    middlewares = require('./middlewares');
+
 module.exports = {
     pre: {
         all: [auth.isAuthenticated]
     },
-    routes: {
-        '/users': {
```

Comment:
I actually think once we figure out some reasonable dependency injection, this may get a lot more clear - we can have a route-builder that marries route configs and middleware beans, and that route-builder could provide lookup services such as the route config for link building.